alibi_detect/utils/pytorch/kernels.py

```diff
@@ -68,16 +68,24 @@ def __init__(
     def sigma(self) -> torch.Tensor:
         return self.log_sigma.exp()
 
-    def forward(self, x: Union[np.ndarray, torch.Tensor], y: Union[np.ndarray, torch.Tensor],
-                infer_sigma: bool = False) -> torch.Tensor:
+    def forward(self, x: Union[np.ndarray, torch.Tensor],
+                y: Union[np.ndarray, torch.Tensor],
+                infer_sigma: bool = False,
+                pairwise: bool = True) -> torch.Tensor:
 
         x, y = torch.as_tensor(x), torch.as_tensor(y)
-        dist = distance.squared_pairwise_distance(x.flatten(1), y.flatten(1))  # [Nx, Ny]
+        if pairwise:
```

Comment:
Check my comment in `distance.py` which might make this  `if else` redundant and reduce it to a kwarg of the distance function.

Reply:
Fixed, now as part of the `squared_pairwise_distance` function argument.